src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -114,6 +142,8 @@ protected function doGetArguments(Request $request, $controller, array $paramete
                 $arguments[] = $request;
             } elseif ($param->isDefaultValueAvailable()) {
                 $arguments[] = $param->getDefaultValue();
+            } elseif ($this->supportsParameterType && null !== ($type = $param->getType()) && $type->allowsNull()) {
```

Comment:
Why not using `ReflectionParameter::allowsNull()` directly?
IIUC, it will work for both `Bar $foo = null` (PHP 5 included) and `?Bar $foo` (PHP 7.1+).